core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -549,8 +558,17 @@ public void writeTo(StreamOutput out) throws IOException {
     public String toString() {
         String sSource = "_na_";
         try {
-            sSource = XContentHelper.convertToJson(source, false);
-        } catch (Exception e) {
+            if (source.length() > MAX_SOURCE_LENGTH_IN_TOSTRING) {
+                // We might slice in the middle of a UTF-8 character here, but BytesRef.utf8ToString() will read more bytes if needed.
+                // There is no risk of ArrayIndexOutOfBoundsException either as source is assumed to be properly UTF-8 encoded.
+                sSource = source.slice(0, MAX_SOURCE_LENGTH_IN_TOSTRING).utf8ToString() + "..." +
```

Comment:
@bleskes was talking below about using only the index, source, and id for the task description. The toString change would still be nice, but less important then. That seems like a good choice to me.